design/secrets.md

```diff
@@ -98,9 +177,26 @@ Multiple credential handling will only be used in the case where a particular BS
 
 Although the handling of secrets will be similar to how credentials are currently managed within Velero, care must be taken to ensure that any new code does not leak the contents of secrets, for example, including them within logs.
 
+## Parallelism
+In order to support parallelism, Velero will need to be able to use multiple credentials simultaneously with the
+ObjectStore.  Currently backups are single threaded and a single BSL will be used throughout the entire backup.  The only
+existing point of parallelism is when a user downloads logs for a backup.  In the current code, `download_request_controller.go`
+creats a new plug-in manager and hence another ObjectStore plugin in parallel with the ObjectStore plugin servicing a
+backup or restore (if one is running).
```

Comment:
My understanding from looking at the instantiation of ObjectStores/Plugin Managers is that each controller is [given a function to create a new plugin manager](https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/server/server.go#L564-L566) which is called every time a controller needs to create an ObjectStore/VolumeSnapshotter. If I'm following the code correctly, this means that there is no process reuse between controllers using the same plugin but operating on different BSLs at a given moment, so current parallelism support should not be impacted. For a controller using the same plugin for different BSLs and VSLs (e.g. MinIO for Objects and AWS for snapshots), the fact that these are initialised independently with different config should mean that this use case will be supported.